### DISCUSSION TITLE
can't use vimium in Firefox new tabs

### DISCUSSION THREAD

**Discussion by H0radricCube** [gdh1995/vimium-c#888] [Q&A]
I'm using firefox on Manjaro. When firefox is on a new tab, vimium-c is not working properly. 
I can't use `x` to close tab nor use `o` bring up the Vomnibar.  

How can i fix this?

**Comment by ghost** [comment]
I'm not sure if this works on Firefox, but if you go to the url [chrome://flags](url) in Chromium based browsers, you can enable the browser flag called "allow extensions in chrome:// urls" and restart the browser. Note that this might work directly with Firefox because the "chrome" in the url refers to the concept of "browser chrome" (a term used to describe exposed backend ui of the browser, I think) and not the Google Chrome browser itself.

**Reply by H0radricCube** [reply]
Thanks for the reply! 
It seems that firefox works not quite the same as chromium.  chrome:// pages like "chrome://newtab" does not load on firefox. I suppose chrome:// counterparts here are about: pages, and they're also blocking any add-ons by design (https://www.reddit.com/r/firefox/comments/ekryjr/comment/fddcmg3/?utm_source=share&utm_medium=web2x&context=3). Still, I didn't find any luck bypassing this restriction.

But I did find a solution XD!

**Comment by H0radricCube** [comment]
The simple solution is to
1. disable all your new-tab extensions
2. install the author's another extension "newtab-adapter" and allow it to taker over your new-tab

In the vimium-c preference settings, I found an item"Allow list of other extension IDs".
Is it possible to add other new-tab extensions' ID here and do the trick? 
Where can I find those IDs (it's not obvious on addons.mozilla.org)?

**Reply by gdh1995** [reply]
You may find extension IDs on `about:debugging#/runtime/this-firefox` (Firefox does hide it on most pages):
![image](https://user-images.githubusercontent.com/5547703/226191208-78f256ce-5575-4f30-bce5-cfd244e3f95b.png)

But, the list is only for Vimium C itself, and can not affect other NewTab extensions.

To make Vimium C's key mappings work on another extension, you'll need to modify the extension's files manually, following the trick in https://github.com/gdh1995/vimium-c/wiki/Inject-into-other-extensions .
